Create new page to navigate to how to join slack channels 7988

kdaca19xx opened the pull request:
<!--  Important! Add the number of the issue you worked on  --> 
Fixes #7988

### What changes did you make?
  - Created `join-slack-channel.html`.
  - Replaced code in `pages/communities-of-practice.html`.
  - Created a function to save the Slack channel URL to `local storage` when user clicks Join Slack Channel button in `communities-of-practice.html`.
  - Created a function to get the Slack channel URL from `local storage` and redirect the users to the correct Slack Channel when users click on the Join Slack Channel button in `join-slack-channel.html`.

### Why did you make the changes (we will use this info to test)?
To enhance the user experience by creating a clear pathway for both existing members and non-members. The dedicated page ensures existing members can easily join Slack channels from the Communities of Practice page, while guiding non-members to first join HfLA before accessing those channels.

<h3>CodeQL Alerts</h3>


After the PR has been submitted and the resulting GitHub actions/checks have been completed, developers should check the PR for CodeQL alert annotations.


<details><summary>Check the PR's comments. If present on your PR, the CodeQL alert looks similar as shown</summary>
  
![Screenshot 2024-10-28 154514](https://github.com/user-attachments/assets/ea66c586-c14c-45fd-8705-1c116224e704)


</details>

Please let us know that you have checked for CodeQL alerts. **Please do not dismiss alerts.**
- [x] I have checked this PR for CodeQL alerts and none were found.
- [ ] I found CodeQL alert(s), and (select one):
   - [ ] I have resolved the CodeQL alert(s) as noted
   - [ ] I believe the CodeQL alert(s) is a false positive (Merge Team will evaluate)
   - [ ] I have followed the Instructions below, but I am still stuck (Merge Team will evaluate)

<details><summary>Instructions for resolving CodeQL alerts</summary>

If CodeQL alert/annotations appear, refer to [How to Resolve CodeQL alerts](https://github.com/hackforla/website/issues/6463#issuecomment-2002573270).  

In general, CodeQL alerts should be resolved prior to PR reviews and merging

</details>

### Screenshots of Proposed Changes To The Website (if any, please do not include screenshots of code changes)
<!-- Notes: 
  - If there are no visual changes to the website, delete all of the script below and replace with "- No visual changes to the website"
  - If there are visual changes to the website, include the 'before' and 'after' screenshots below. 
  - If your images are too big, use the <img src="" width="" length="" />  syntax instead of ![image](link) to format the images
  - If images are not loading properly, you might need to double check the syntax or add a newline after the closing </summary> tag 
 --> 

<details>
<summary>Visuals before changes are applied</summary>

There was no `join-slack-channel.html` page, so there's no visual to share.

</details>

<details>
<summary>Visuals after changes are applied</summary>
 
<img width="1600" height="900" alt="after_Issue 7988_data science" src="https://github.com/user-attachments/assets/cf847a71-6ffd-4f8b-abed-d632ccf1189a" />

<img width="1600" height="900" alt="after_Issue 7988_engineering" src="https://github.com/user-attachments/assets/7546720a-8937-46b0-9e4b-5a3e9200b501" />

<img width="1600" height="900" alt="after_Issue 7988_Ops" src="https://github.com/user-attachments/assets/0d0a9732-8a92-451c-bbb5-16565e16058e" />

<img width="1600" height="900" alt="after_Issue 7988_project and product mgmt" src="https://github.com/user-attachments/assets/53bc6551-55de-4a9b-bf95-cd2a6389d2f8" />

<img width="1600" height="900" alt="after_Issue 7988_UI and UX design" src="https://github.com/user-attachments/assets/e1be9db0-fedc-440e-8318-ba134a84f89d" />



</details>

kdaca19xx committed:
modified and new html pages

kdaca19xx committed:
Merge branch 'gh-pages' of https://github.com/hackforla/website into create-new-page-to-navigate-to-how-to-join-slack-channels-7988

kdaca19xx committed:
latest changes to issue 7988

kdaca19xx committed:
Merge branch 'gh-pages' of https://github.com/hackforla/website into create-new-page-to-navigate-to-how-to-join-slack-channels-7988

kdaca19xx committed:
changes to 7988

kdaca19xx committed:
Merge branch 'gh-pages' of https://github.com/hackforla/website into create-new-page-to-navigate-to-how-to-join-slack-channels-7988

kdaca19xx committed:
I deleted files

caz002 commented:
Review ETA: 8/25/25
Availability: after 7pm on Weekdays

caz002 reviewed (commented):
Thank you for working on this issue @kdaca19xx !

# Things Done Well
- PR description accurately describes changes and why they were made
- Changes appear on the website and line up with what was asked in the instructions
# Things to Change
-  In #7988, I noticed the last bullet point says to add a new function to `assets/js/communities-of-practice.js` and not `assets/js/join-slack-channel.js`. I'm not sure if the bullet point is a typo though, so let me know if you were instructed to create an `assets/js/join-slack-channel.js` file, and I can approve your issue.
- **Edit:** There is also a CodeQL error for "Memory Activity Trigger" that popped up after I reviewed the PR, I'm not sure if it was because of your code though. I messaged Will to ask for more information.

DDVVPP commented:
Looks great @kdaca19xx !
- The PR is done with the correct branch
- The PR title accurately describes the issue
- There is a linked issue
- The visuals for the added html page are clear
- All of the links are directing the user to the correct routes/slack channels

A Few Additional Comments
- This is not required for approval, but I wonder if you would want to center all of the text on the page for a more pleasing visual aesthetic? Currently it looks like the <p> text is left-aligned
- It would also be nice to see a little space below the 'Join Slack Channel' button in the newly created page - again not required for approval!
- Agreeing with @caz002's comment - it looks like you've added a new `join-slack-channel.js` file rather than adding the new function to the existing `communities-of-practice.js` file.  Happy to approve once this is resolved!


Thank you!

DDVVPP reviewed (commented):
Looks great @kdaca19xx !
- The PR is done with the correct branch
- The PR title accurately describes the issue
- There is a linked issue
- The visuals for the added html page are clear
- All of the links are directing the user to the correct routes/slack channels

A Few Additional Comments
- This is not required for approval, but I wonder if you would want to center all of the text on the page for a more pleasing visual aesthetic? Currently it looks like the `<p>` text is left-aligned
- It would also be nice to see a little space below the 'Join Slack Channel' button in the newly created page - again not required for approval!
- Agreeing with @caz002's comment - it looks like you've added a new `join-slack-channel.js` file rather than adding the new function to the existing `communities-of-practice.js` file.  Happy to approve once this is resolved!


Thank you!

kdaca19xx commented:
Hi @caz002 and @DDVVPP,

The two ways I've added the `join-slack-channel.js` code to `communities-of-channel.js` file is causing an issue with the data science button. Here are notes on what I tried. Hope they're clear:

1. Pasted all of the code from `join-slack-channel.js` directly under the code in `communities-of-practice.js`. (I know.)
2. Changed the `<script>` reference in `join-slack-channel.html` to use `communities-of-practice.js`.

Result: All buttons worked except the Data Science button. On load it said "return to communities". When clicked it changed to "join slack channel", and when that was clicked it went to the Data Science group on Slack.

3. Removed the `join-slack-channel.js` code I had pasted into `communities-of-practice.js` (using ctrl+z).
4. Changed the `<script>` reference in `join-slack-channel.html` back to `join-slack-channel.js`.

Result: Everything went back to normal.

5. Added lines 5–49 from `join-slack-channel.js` into `communities-of-practice.js` (on line 90, before the final `});`).
6. Changed the `<script>` reference in `join-slack-channel.html` to use `communities-of-practice.js`.

Result: Same issue as before — all buttons worked except the Data Science button. On load it said "return to communities". When clicked it said "join slack channel", and when that was clicked it went to the Data Science group on Slack.

7. Removed lines 5–49 (from `join-slack-channel.js`) out of `communities-of-practice.js`.
8. Changed the `<script>` reference in `join-slack-channel.html` back to `join-slack-channel.js`.

Result: Everything went back to normal.

I'll reach out to a lead on Monday. Thx for your patience!

kdaca19xx committed:
requested changes

kdaca19xx committed:
Merge branch 'gh-pages' of https://github.com/hackforla/website into create-new-page-to-navigate-to-how-to-join-slack-channels-7988

caz002 reviewed (commented):
Hi @kdaca19xx,

Thank you for making the changes! The code looks great, and the website works as expected. I wanted to add that if you're still having issues with removing `join-slack-channels.js`, and your using a code editor like VS Code, you can delete the file manually by right clicking on it and selecting the option, and then you can use `git add .` and the regular procedure you've used previous times to update the branch. I think `git add .` checks for any changes, both additions and removals, so it should be okay to still use.

kdaca19xx committed:
deleted join-slack-channel.js

caz002 reviewed (approved):
Looks great, just approved!
